### DISCUSSION TITLE
How to set up a dynamic route using front matter

### DISCUSSION THREAD

**Discussion by nemu1113** [kwchang0831/svelte-QWER#29] [Q&A] [answered]
Hello!
How would you prefer to set up a dynamic route using the front matter (title, published) of a markdown file?

When adding article files according to docs instructions, the specification generates paths on a directory basis, so I believe some additional modification is necessary to achieve the above, but I would like to set up a dynamic route only for a given folder (see below), while also retaining the directory-based generation logic.

**Current**
- directoryPath: user/blogs/posts/(md-filename).md
- url: /posts/(md-filename)

**After Changed**
- directoryPath: user/blogs/posts/(post year)/(md-filename).md
- url: /posts/(post-year)/(post-month)/(post-date)/(title)

Any good ideas would be greatly appreciated.
Thank you in advance.

**Comment by kwchang0831** [comment]
Currently QWER does not support this kind of routing rule.

It is trivial to do I think. Just that my code is a bit messy.

I will find time to do what you just described if that's the feature you are requesting.

**Comment by nemu1113** [comment]
@kwchang0831
Thank you for your reply.
I understand that it is an unsupported feature.

Well, my blog has more than 300 articles in the past, therefore considering future maintainability, I would like to control it with front matter if possible.

I'm still trying to figure out if there's an easier way to do this.

**Reply by kwchang0831** [reply]
300 aritcles. A lof of work you have done. :D Nice.

To support what you described, what I will do would be...
1. Create a special rule to detect whether the path is `/posts/{year}/{title}.{file-exntension}`
1. Transfrom the output slug / url to `/posts/{year}/{month}/{day}/{title}` instead of `/post/{year}/{title}`.

The rest of routing rule would stay the same, only the `/posts/` folder has this rule.

And then, the on/off trigger will put inside `QWER.config.js`, so you don't have to modify your data's front matter.

**Comment by kwchang0831** [comment, marked as answer]
@nemu1113 Your request has been fulfilled as of (63efeeb4dcaddad4fd85bd2bcb6329839bc3220b).

The doc is also updated: https://docs-svelte-qwer.vercel.app/config/qwer-config-js#special-routing-rules

Migrate your `QWER.config.js`.

```diff
export const UserConfig = {
  ...
+  RoutingRules: {
+    PostsYearMonthDate: true,
+  },
};
```

Let me know if there is any issues.

Also, I have not yet tested 300+ articles. Hopefully QWER is able to handle that many posts. I suspect the performance of index page will degrade. Just not sure by how much. 

However, I expect PWA offline feature to make consecutive visits work as a charm without delaying while loading the posts. Hmmm...let's see... (finger crossed).

**Reply by nemu1113** [reply]
@kwchang0831 

Thank you for your response regarding this enhancement.

I would like to close [issue#33](https://github.com/kwchang0831/svelte-QWER/pull/33#issue-1385413851) after the following is resolved.
I have confirmed that the modified code is functioning properly.
